### DISCUSSION TITLE
Plot function for distributions or histograms

### DISCUSSION THREAD

**Discussion by dzolg** [sqlitebrowser/sqlitebrowser#3311] [Ideas]
Hi all,

awesome program. 
I understand that the program shall be as sleek as possible but a feature that I am missing a lot is a density/histogram plot to get a feel for the data in a single column. Basically a way to very quickly check the distribution of values in a numeric column via the plot function. Either via density plot or a histogram plot or so fourth.

Any thoughts?
Cheers

**Comment by justinclift** [comment]
Hmmm, any kind of graphic or screenshot or similar you can paste here for reference?

Something to illustrate what it'd ideally show?

**Reply by dzolg** [reply]
In my work I have a lot of high throughput omics data that gets stored in sqlite. I would be looking for something like this to quickly check the distribution of values in a column. Quick and dirty R example from a column that has a large number of measurement values around somewhat centered around 0.

Histogram
![image](https://user-images.githubusercontent.com/40891309/229124509-ae996bd5-5bb0-4319-b0e1-af43bb19b9bd.png)
Density
![image](https://user-images.githubusercontent.com/40891309/229124620-d6141857-447b-4e20-83d7-171d84b721ca.png)

**Reply by chrisjlocke** [reply]
Could you not create a view based on the count?

![image](https://user-images.githubusercontent.com/7872218/229128693-027fba7b-8429-4b2f-9893-800e27545e0c.png)

So looking at a table of hotel bookings, most people stay 1 night .. tailing off as the nights increase .. apart from 14 nights (which is expected)

You should be able to point the plot at this view?

**Reply by dzolg** [reply]
Generally a good idea, just not very practical for floats/doubles (e.g. see example above, approx 8k unique values, hence the binning/density calculation). In the end, I can do extensive plotting in R; I just thought it would be awesome to have a quick look where you don't have to define own bins etc.

**Comment by justinclift** [comment]
Ahhh, yep.  Fairly standard examples of it.  Shouldn't be too hard to implement, it just needs one of our developers to have a burst of enthusiasm / interest for it. :smile:

**Comment by mfrejno** [comment]
This would be very useful indeed!

**Comment by Rscho314** [comment]
I just wanted to +1 this request for enhancement. Having a density plot readily available would be invaluable for eyeballing data.
For the record, I am an epidemiologist doing medical research, and I make heavy use of SQLiteBrowser. I think I am far from being the only one with that use case. Unfortunately, I am also nowhere near capable enough in C++ to walk the walk, so I'll just talk the talk! :-)
